Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/intersection.h

```diff
@@ -26,9 +26,1517 @@
+                   typename boost::has_range_iterator<
+                    typename boost::mpl::eval_if<
+                      boost::has_range_iterator<PolylineRange>,
+                      boost::range_value<PolylineRange>,
+                      boost::false_type >::type
+                   >::type
+                  >::type* = 0//end enable_if
+    )
+{
+  CGAL_precondition(CGAL::is_triangle_mesh(mesh));
+
+  return do_intersect(mesh, polylines, parameters::all_default());
+}
+
+
+template <class TriangleMesh
+          , class Polyline>
+bool do_intersect(const TriangleMesh& mesh,
+                  const Polyline& polyline,
+                  const typename boost::disable_if<
+                  typename boost::has_range_const_iterator<TriangleMesh>::type
+                  >::type* = 0
+    ,const typename boost::disable_if<
+                  typename boost::has_range_iterator<
+                  typename boost::mpl::eval_if<
+                  boost::has_range_iterator<Polyline>,
+                  boost::range_value<Polyline>,
+                  boost::false_type >::type
+                  >::type
+                  >::type* = 0
+                  )
+{
+  CGAL_precondition(CGAL::is_triangle_mesh(mesh));
+
+  return do_intersect(mesh, polyline, parameters::all_default());
+}
+}//end PMP
+namespace internal{
+
+template<class TriangleMeshRange
+         , typename OutputIterator
+         , class NamedParametersRange
+         , class NamedParameter
+         >
+struct Mesh_callback
+{
+  typedef typename boost::range_value<TriangleMeshRange>::type TriangleMesh;
+  typedef typename GetGeomTraits<TriangleMesh, NamedParameter>::type GT;
+  typedef typename boost::range_value<NamedParametersRange>::type VertexPointMapNP;
+  typedef typename GetVertexPointMap<TriangleMesh, VertexPointMapNP>::const_type VPM;
+  typedef CGAL::AABB_face_graph_triangle_primitive<TriangleMesh, VPM> Primitive;
+  typedef CGAL::AABB_traits<GT, Primitive> Traits;
+  typedef CGAL::AABB_tree<Traits> AABBTree;
+  typedef typename boost::graph_traits<TriangleMesh>::face_descriptor face_descriptor;
+
+  // todo : add a vector/map of trees for each mesh.
+  // fill them in the operator and test for inclusion with
+  // Side_of_triangle_mesh.
+  const TriangleMeshRange& meshes;
+  OutputIterator m_iterator;
+  const bool& test_volume;
+  const NamedParametersRange& nps;
+  const NamedParameter& np;
+  std::vector<AABBTree*> trees;
+
+  std::vector<std::vector<CGAL::Point_3<GT> > > points_of_interest;
+
+  Mesh_callback(const TriangleMeshRange& meshes,
+                OutputIterator iterator,
+                const bool& test_volume,
+                const NamedParametersRange& nps,
+                const NamedParameter& np)
+    : meshes(meshes), m_iterator(iterator),
+      test_volume(test_volume), nps(nps), np(np)
+  {
+    int size = std::distance(meshes.begin(), meshes.end());
+    trees = std::vector<AABBTree*>(size, NULL);
```

Comment:
if you don't have a destructor, it will leak!

Reply:
Holly Molly ! Good call !